docs/architecture/adr/ADR044-dev-namespaces.md

```diff
@@ -0,0 +1,74 @@
+# ADR044: Dev namespaces
+
+## Status
+
+Pending
+
+## Context
+
+Following the retirement of gsp-local the service teams have no way of testing
+changes before a merge to master. So the feedback cycle is very slow, and
+potentially risky. We need to enable devs to get more rapid feedback through use
+of namespaces that allow for deployments from arbitrary branches or from local
+`kubectl apply` actions.
+
+This presents several security problems that need to be addressed.
+
+The in-cluster concourse:
```

Comment:
so in-cluster concourse is becoming untrusted now?

Reply:
If that's how you want to interpret this, yes. At the moment it can make cluster-level changes (if configured accordingly via various switches in `x-cluster-config`), which this ADR will remove. If we want to elevate service team members' access then we need to reduce the in-cluster concourse's permissions to the same ceiling as service-team members.